overlord/hookstate/context.go

```diff
@@ -25,16 +25,18 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/snapcore/snapd/overlord/configstate/transaction"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 )
 
 // Context represents the context under which a given hook is running.
 type Context struct {
-	task    *state.Task
-	setup   *HookSetup
-	id      string
-	handler Handler
+	task      *state.Task
+	setup     *HookSetup
+	id        string
+	handler   Handler
+	finalizer func() error
```

Comment:
This needs to be something like:
`onDone []func() error`
It needs to be a list, as anyone else must be able to do the same thing on this context for their own reasons. We also need to call out that it's on done, specifically, as we may need onError soon.